src/core/json/json_like.rs

```diff
@@ -41,6 +43,66 @@ pub trait JsonLike<'json>: Sized {
     fn get_path<T: AsRef<str>>(&'json self, path: &[T]) -> Option<&Self>;
     fn get_key(&'json self, path: &str) -> Option<&Self>;
     fn group_by(&'json self, path: &[String]) -> HashMap<String, Vec<&Self>>;
+
+    ///
+    /// Used to apply `input_transforms` operations on the given `json_like`
+    /// struct. Current supported operations are `modify` - rename.
+    fn handle_input_transforms(
```

Comment:
Design Feedback: JsonLike shouldn't make assumptions about how it's being used, in this case — `handle_input_transforms` function has the following problems:
- Depends on a transformation that's defined in IR::model. 
- Only works one level deep.
- Only renames keys, so isn't a general purpose transform.

Reply:
Try something like — 

```rust
enum Transform {
    
    Rename(String) // Renames the provided key in the object
    // TODO: add more operations as time goes on
}


enum Lens {
    Select(String), // Selects the value at the provided key
    ForEach(Rc<Lens>), // Applies the provided lens to each value in 
    Compose(Rc<Lens>, Rc<Lens>), // Composes two lenses together
    Transform(Rc<Lens>, Transform), // Applies a transformation on the lens
}

impl Lens {
    fn transform<'json>(self, J: &mut JsonLike<'json>) -> J {
        todo!()
    }
}
```

Reply:
* Better organized the codebase
* Added a more complex (deep nested) test case to prove that nesting is working on deep nested objects
* Refactored the code to allow for other operations in the future easily